### PR TITLE
[SYCL] [L0] [PI] Corrections to interop tests to ensure batching behavior

### PR DESCRIPTION
**Pull request by rdeodhar**
Some interop tests depend on assumptions about the plugin behavior with regard to commandlist and event usage. The expectations can be met by setting env vars that control batching and device-only events usage.

### CODE REVIEW

File: SYCL/Plugin/interop-level-zero-interop-task-mem.cpp [hunk 1]
```diff
@@ -1,6 +1,6 @@
 // REQUIRES: level_zero, level_zero_dev_kit
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %level_zero_options %s -o %t.out
-// RUN: env SYCL_BE=PI_LEVEL_ZERO %GPU_RUN_PLACEHOLDER %t.out
+// RUN: env SYCL_BE=PI_LEVEL_ZERO SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS=2 SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 %GPU_RUN_PLACEHOLDER %t.out
```
Comment (smaslov-intel) [inline SYCL/Plugin/interop-level-zero-interop-task-mem.cpp:3]: I don't see any check of traces in this test, so it should run with any env settings

File: SYCL/Plugin/interop-level-zero-keep-ownership.cpp [hunk 2]
```diff
@@ -1,6 +1,6 @@
 // REQUIRES: level_zero, level_zero_dev_kit
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out %level_zero_options
-// RUN: env SYCL_BE=PI_LEVEL_ZERO %GPU_RUN_PLACEHOLDER %t.out
+// RUN: env SYCL_BE=PI_LEVEL_ZERO SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS=2 SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 %GPU_RUN_PLACEHOLDER %t.out
```
Comment (smaslov-intel) [inline SYCL/Plugin/interop-level-zero-keep-ownership.cpp:3]: I don't see any check of traces in this test, so it should run with any env settings

File: SYCL/Plugin/interop-level-zero-thread.cpp [hunk 3]
```diff
@@ -1,6 +1,6 @@
 // REQUIRES: level_zero, level_zero_dev_kit
 // RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out %level_zero_options %threads_lib
-// RUN: %BE_RUN_PLACEHOLDER %t.out
+// RUN: env SYCL_PI_LEVEL_ZERO_DEVICE_SCOPE_EVENTS=2 SYCL_PI_LEVEL_ZERO_USE_IMMEDIATE_COMMANDLISTS=0 %BE_RUN_PLACEHOLDER %t.out
```
Comment (smaslov-intel) [inline SYCL/Plugin/interop-level-zero-thread.cpp:3]: same here

**Review by smaslov-intel — CHANGES_REQUESTED** [review]

**Comment by rdeodhar** [comment]
These tests fail if immediate commandlists are used in the L0 plugin.
The tests will need to be revisited when/if use of immediate commandlists is made the default.

**Pull request closed without merging**